clients/mongoStoreClient.go

```diff
@@ -207,18 +212,17 @@ func (d MongoStoreClient) ExecuteQuery(details *model.QueryData) ([]byte, error)
 		Sort(sort).
 		Select(filter).
 		All(&results)
-	if err != nil {
+
+	if err != nil && err != mgo.ErrNotFound {
 		log.Println(fmt.Sprintf("ExecuteQuery: mongo query took [%.5f] but failed with error [%s] ", time.Now().Sub(startQueryTime).Seconds(), err.Error()))
 		return nil, err
+	} else if err != nil && err == mgo.ErrNotFound || len(results) == 0 {
+		log.Println(fmt.Sprintf("ExecuteQuery: mongo query took [%.5f] and found no records", time.Now().Sub(startQueryTime).Seconds()))
+		return []byte("[]"), nil
```

Comment:
yeah possibly a generic error handler for mongo stuff but at this stage I am ok with the repetition